RegistrarApi/Controllers/SettingController.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using System.Web.Http;
+using Common;
+using Common.Exceptions;
+using Registrar.Database.Interfaces;
+using Registrar.Models.Request;
+
+namespace Registrar.Api.Controllers
+{
+    [RoutePrefix("settings")]
+    public class SettingController : ApiController
+    {
+        private readonly IRegiSettingStore _store;
+
+        public SettingController(IRegiSettingStore store)
+        {
+            _store = store;
+        }
+
+        [HttpPost]
+        [Route("")]
+        [ApiKeyAuth]
+        public async Task<IHttpActionResult> Post(UpdateRegiSetting setting)
+        {
+            try
+            {
+                var updatedSetting = await _store.UpdateSetting(setting);
+                return Json(updatedSetting);
+            }
+            catch (RecordNotFoundException)
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpGet]
+        [Route("list")]
+        [ApiKeyAuth]
+        public async Task<IHttpActionResult> List()
+        {
+            try
+            {
+                var settings = await _store.ListSettings();
+                return Json(settings);
+            }
+            catch (RecordNotFoundException)
+            {
+                return NotFound();
```

Comment:
But there is an error - if you're trying to get something and it isn't present you should get a Resource Not Found response, aka 404. 404's refer to any entity related to the request not being found, not just the route. If you then want to handle it not being found by showing an empty list, that's up to the client, not the server to decide. I'm leaving this as `NotFound`.